### ISSUE TITLE
Make sampling methods for loss functions for language modeling

### ISSUE THREAD

**Issue by unnonouno** [chainer/chainer#1791]
There is no sampling methods for language modeling such as hierarchical softmax (HSM) and blackout.
Especially HSM behaves like a chain of binary classifiers. It can sample a word with O(log n) time where n is the vocabulary size.
